### PR TITLE
Add support for publish metadata callback

### PR DESCRIPTION
**Pull request by arobson**
If present, the callback is invoked with a hash that has `activated` and `skipped` counts. This should provide a lot of flexibility for additional metadata or features in the future and is far less awful than the approach I took in PR #107 (may it perish forever in the colorless flame of dev/null :smiling_imp:).

### CODE REVIEW

File: spec/channelDefinition.spec.js [hunk 1]
```diff
@@ -38,4 +38,26 @@ describe( "ChannelDefinition", function() {
...
+	describe( "when publishing and subscribing directly on the channel with a callback", function() {
+		it( "should provide correct counts for both activated and skipped subscriptions", function() {
+			var channel = postal.channel( "Dalek" );
+			var cb = function() {};
+
+			var sub1 = channel.subscribe( { topic: "exterminate", callback: cb } ).constraint( function( x ) {
+				return x.target !== "Dalek";
+			} );
+
+			channel.publish( "derpxterminate", { target: "The Doctor" }, function( result ) {
```
Comment (ifandelse) [inline spec/channelDefinition.spec.js]: nice. DERPINAAAAATE!

Which also makes me think of this:

**Comment by avanderhoorn** [comment]
Just wondering what sort of use cases this is designed for?

**Comment by ifandelse** [comment]
@avanderhoorn If I recall correctly, @arobson was using it this feature in [wascally](https://github.com/leankit-labs/wascally) (which takes a dependency on postal). Wascally checks to verify that a message (coming in from rabbit and then published locally in postal) is handled by at least one subscriber. If not, the message isn't acknowledged, and the broker will re-deliver to a different consumer.

**Comment by avanderhoorn** [comment]
@ifandelse Cool thanks! Interesting use case.

**Pull request merged**